Realm/RLMResults.h

```diff
@@ -20,6 +20,25 @@
 
 RLM_HEADER_AUDIT_BEGIN(nullability, sendability)
 
+/// A block type used for APIs which asynchronously return a `Results`.
+typedef void(^RLMResultsCompletionBlock)(RLMResults * _Nullable, NSError * _Nullable);
+
+/**
+ Determines wait for download behavior when subscribing on RLMResults.
+ @see ``[RLMResults subscribeWithName:waitForSyncMode:onQueue:completion:]``
+*/
+typedef NS_ENUM(NSUInteger, RLMWaitForSyncMode) {
+    /// `subscribeWithName` will return once matching objects are downloaded
+    /// from the server only when the subscription is created the first time. If the
+    /// subscriptions already exists, the method will return without waiting for new downloads.
```

Comment:
`subscribeWithName:` always returns immediately and what this controls is when the callback is invoked.